src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
I'm not sure if I understand you... How would this solve the dilemma between "correct value to have *almost* the requested value" vs. "either go with requested value, or signal error".